burn-core/src/nn/initializer.rs

```diff
@@ -14,7 +15,8 @@ pub enum Initializer {
     Constant(f64),
     Ones,
     Zeros,
-    // TODO: add Xavier initialization
+    XavierUniform(f64),
```

Comment:
Since you're working on this. Could we add public documentation to the Initializer and its variants? This is for completeness. 

(sorry, it's actually my fault for not adding in the first place when I implemented it)

Reply:
Of course!
In fact I shall refactor further, because I notice a strong similarity between the new Xavier init and the default initialization of linear and conv

Reply:
> Of course! In fact I shall refactor further, because I notice a strong similarity between the new Xavier init and the default initialization of linear and conv

Yeah, you also probably noticed a peculiar initializer called UniformDefault (I made it up and @nathanielsimard could not come up with a better alternative name at the time). If it's the Xavier, then we should refactor as well. That would be great.

Reply:
Yes, UniformDefault is exactly what I'm changing. It's not exactly Xavier but it is similar. I will migrate this initialization in the initializer so the caller won't have to do it itself

Reply:
Maybe it's one of those: https://pytorch.org/docs/stable/nn.init.html but I couldn't figure out.